### PR TITLE
feat: implement the ability to configure release rules

### PR DESCRIPTION
**Pull request by acazacu**
Add the ability to pass custom release rules to Semantic Release. If no rules are provided the following defaults will be used:

```typescript
interface ReleaseRule {
  release: string | false;
  scope?: string;
  subject?: string;
  type?: string;
}

export const DEFAULT_RELEASE_RULES: ReleaseRule[] = [
  { release: 'patch', type: 'build' },
  { release: 'patch', type: 'chore' },
  { release: 'patch', type: 'ci' },
  { release: 'patch', type: 'docs' },
  { release: 'patch', type: 'improvement' },
  { release: 'patch', type: 'refactor' },
];
```

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -53,6 +53,17 @@ steps:
       # Attach the new line separated listed glob patterns to the release.
       release-assets: |
         ./generated/my-asset.tar.gz
+      # Configure the semantic release branches parameter: https://semantic-release.gitbook.io/semantic-release/usage/workflow-configuration#branches-properties
+      # If not specified semantic-release will use its default branches
+      # configuration, as specified in their Javascript API documentation:
+      # https://github.com/semantic-release/semantic-release/blob/master/docs/usage/configuration.md#branches
+      release-branches: '["+([0-9])?(.{+([0-9]),x}).x","master","next","next-major",{"name":"beta","prerelease":"beta"},{"name":"alpha","prerelease":"alpha"}]'
```
Comment (mad-it) [inline README.md:60]: Is this nessesary?

File: src/index.ts [hunk 2]
```diff
@@ -50,19 +36,21 @@ export const release = async (): Promise<void> => {
...
     plugins: generatePlugins({
-      commitAssets: parseInputCommitAssets(),
-      isNodeModule: parseInputNodeModule(),
-      releaseAssets: parseInputReleaseAssets(),
+      commitAssets: processInputCommitAssets(),
+      isNodeModule: processInputNodeModule(),
+      releaseAssets: processInputReleaseAssets(),
     }),
-    releaseRules: releaseRulesExtension,
+    preset: 'angular',
```
[the remark this answers is not in the document]
Comment (oleg-koval) [reply src/index.ts:52]: Looks like this setting cant be configurable as we are using Angular Commit Message Conventions. To use other (atom, codemirror, ember, eslint, express, jquery, jshint, conventionalcommits) we should also prepare parsers for each style.

File: src/index.ts [hunk 3]
```diff
@@ -62,7 +49,8 @@ export const release = async (): Promise<void> => {
       isNodeModule: processInputNodeModule(),
       releaseAssets: processInputReleaseAssets(),
     }),
-    releaseRules: releaseRulesExtension,
+    preset: 'angular',
```
Comment (mad-it) [inline src/index.ts:52]: I know that "we" (Dott) use the 'angular' preset, but is this something that should be hardcoded? This removes the flexibility of using this release-me action with other commit messages.

File: README.md (hunk 1 above)
Comment (acazacu) [reply README.md]: It is the default value...

File: src/index.ts (hunk 3 above)
Comment (acazacu) [reply src/index.ts:52]: https://github.com/ridedott/release-me-action/pull/197#discussion_r440015993

We can do that in a separate PR?

File: README.md (hunk 1 above)
Comment (mad-it) [reply README.md]: I was curious why this was part of this PR.

File: src/index.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (merlinnot) [reply src/index.ts:52]: I agree, let's add this to the documentation and fix it in a separate PR to better support the Community.

File: src/utilities/inputProcessors.spec.ts [hunk 4]
```diff
@@ -1,11 +1,13 @@
 import * as actionsCore from '@actions/core';
 
 import {
+  DEFAULT_RELEASE_RULES,
```
[the remark this answers is not in the document]
Comment (acazacu) [reply src/utilities/inputProcessors.spec.ts]: Resolved!

File: src/index.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (acazacu) [reply src/index.ts:52]: This information is already present in the README.md file, line 15 😬 .

File: README.md (hunk 1 above)
Comment (acazacu) [reply README.md]: Oh, good catch! This was a slip, resolved!

**Pull request merged**